web/app.js

```diff
@@ -1410,7 +1433,18 @@ const PDFViewerApplication = {
 
     const dispatchEventName = generateRandomStringForSandbox(objects);
     const calculationOrder = [];
-    scripting.createSandbox({ objects, dispatchEventName, calculationOrder });
+    scripting.createSandbox({
+      objects,
+      dispatchEventName,
+      calculationOrder,
+      docInfo: {
+        ...info,
+        metadata,
+        filename: contentDispositionFilename,
+        numPages: pdfDocument.numPages,
+        URL: window.location.href,
```

Comment:
Does this actually do what you want/expect across different types of PDF.js builds?